src/gensys.jl

```diff
@@ -88,23 +89,28 @@ end
 #- gensys methods -#
 ## -------------- ##
 
+# Wrapper methods ensuring no side effects on input arguments
+gensys(Γ0, Γ1, c, ψ, π) = gensys!(copy(Γ0), copy(Γ1), copy(c), copy(ψ), copy(π))
+gensys(Γ0, Γ1, c, ψ, π, div) = gensys!(copy(Γ0), copy(Γ1), copy(c), copy(ψ), copy(π), div)
```

Comment:
Do all these need to be copied? Meaning do all of them get mutated within `gensys!`?